tests/unit/resources/test_fcsfile.py

```diff
@@ -79,6 +81,31 @@ def test_gets_file_internal_compensation(ENDPOINT_BASE, client, fcs_files, spill
     assert type(comp) == Compensation
 
 
+@responses.activate
+def test_throws_correct_error_when_no_file_internal_compensation(
+    ENDPOINT_BASE, client, fcs_files
+):
+    # Given: An FcsFile with no spill string
+    file_data = fcs_files[0]
+    file_data["spillString"] = ""
+    file = FcsFile.from_dict(file_data)
+    expected_response = fcs_files[0].copy()
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{file._id}",
+        json=expected_response,
+    )
+
+    # When:
+    with pytest.raises(ValueError) as err:
```

Comment:
Not a python person: what raises this? Something automatically because it's an empty string?

Reply:
This is `pytest` expecting a `ValueError` to be raised in the test; `.get_file_internal_compensation` throws a `ValueError` when `.has_file_internal_comp` is `False`, so this asserts that the correct error (and not some other `ValueError`) is thrown.

This _could_ throw a warning or simply return `None`, instead of an error; what do you think?

Reply:
Ah sorry, I was looking at `def spill_string` not `def get_file_internal_compensation`.